experimental/dds/tree2/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -158,62 +146,88 @@ function rebaseMarkList<TNodeChange>(
 		// Note that we first add lineage for `baseMark` to `lineageRecipients`, then handle adding lineage to `rebasedMark`,
 		// then add `baseMark` to `lineageEntries` so that `rebasedMark` does not get an entry for `baseMark`.
 		// `rebasedMark` should already have a detach event for `baseMark`.
-		if (markEmptiesCells(baseMark)) {
-			assert(isDetachMark(baseMark), 0x709 /* Only detach marks should empty cells */);
-			const detachId = getDetachCellId(baseMark, baseIntention);
+		if (markEmptiesCells(baseMark) || isTransientEffect(baseMark)) {
+			const detachId = getOutputCellId(baseMark, baseRevision, metadata);
+			assert(detachId !== undefined, "Mark which empties cells should have a detach ID");
 			assert(detachId.revision !== undefined, 0x74a /* Detach ID should have a revision */);
-			addLineageToRecipients(lineageRecipients, detachId.revision, detachId.localId, length);
-		}
+			addLineageToRecipients(
+				lineageRecipients,
+				detachId.revision,
+				detachId.localId,
+				baseMark.count,
+				metadata,
+			);
 
-		if (areInputCellsEmpty(rebasedMark)) {
-			if (markEmptiesCells(baseMark)) {
-				assert(isDetachMark(baseMark), 0x74b /* Only detaches empty cells */);
-				if (baseMark.type === "MoveOut" || baseMark.detachIdOverride === undefined) {
-					setMarkAdjacentCells(rebasedMark, detachBlock);
-				}
+			if (isInverseAttach(baseMark)) {
```

Comment:
Why do we care that baseMark is an inverse attach?

Reply:
Suppose in changeset A we empty some cell X. The new ID for cell X will contain `adjacentCells` describing any adjacent cells which were emptied in changeset A. Now suppose in a subsequent changeset B we fill cell X. If we invert changeset B, we want to return cell X's ID to what it was after changeset A was applied, including the `adjacentCells` from A. When a change targeting cell X rebases over B-inverse we want to make sure to set its `adjacentCells` field to the value stored in B's `detachIdOverride` instead of deriving new `adjacentCells` based on detaches in B.

Reply:
There were previously other reasons that we treated inverse attaches differently here, but after reviewing this I've simplified the logic somewhat.